aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/CredentialProviders.kt

```diff
@@ -42,36 +55,59 @@ class CredentialsProviderDecorator : ClientCodegenDecorator {
         listOf(
             SdkConfigSection.create { section ->
                 writable {
-                    rust("${section.serviceConfigBuilder}.set_credentials_provider(${section.sdkConfig}.credentials_provider().clone());")
+                    rust("${section.serviceConfigBuilder}.set_credentials_provider(${section.sdkConfig}.credentials_provider().cloned());")
```

Comment:
`clone` vs `cloned` always gets me. I wish they had given it a more distinct name.